app/code/Magento/InventoryInStorePickup/Model/Source/Validator/PhoneValidator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickup\Model\Source\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Model\SourceValidatorInterface;
+use Magento\InventoryInStorePickup\Model\Source\GetIsPickupLocationActive;
+
+/**
+ * Check if phone is set for Pickup Location.
+ */
+class PhoneValidator implements SourceValidatorInterface
+{
+    /**
+     * @var ValidationResultFactory
+     */
+    private $validationResultFactory;
+
+    /**
+     * @var GetIsPickupLocationActive
+     */
+    private $getIsPickupLocationActive;
+
+    /**
+     * @param ValidationResultFactory $validationResultFactory
+     * @param GetIsPickupLocationActive $getIsPickupLocationActive
+     */
+    public function __construct(
+        ValidationResultFactory $validationResultFactory,
+        GetIsPickupLocationActive $getIsPickupLocationActive
+    ) {
+        $this->validationResultFactory = $validationResultFactory;
+        $this->getIsPickupLocationActive = $getIsPickupLocationActive;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function validate(SourceInterface $source): ValidationResult
+    {
+        $value = (string)$source->getCity();
+        $errors = [];
+
+        if ($this->getIsPickupLocationActive->execute($source) && '' === trim($value)) {
```

Comment:
```suggestion
        if ($this->getIsPickupLocationActive->execute($source) && !trim($value)) {
```

Reply:
Or at least `trim($value) === ''` instead of `'' === trim($value)`

Reply:
Excuse me what problem does it solve?
I see the same declarations in other validators.

Reply:
Just seek for places where `value` is placed on the left, and a `$variable` - on the right, not vice-versa.
Except `\Magento\InventoryConfiguration\Model\GetStockItemConfiguration`.